graphql_api/types/test_results_aggregates/test_results_aggregates.py

```diff
@@ -1,81 +1,153 @@
-from typing import TypedDict
+from dataclasses import dataclass
 
+import polars as pl
 from ariadne import ObjectType
 from graphql import GraphQLResolveInfo
+from shared.django_apps.core.models import Repository
 
-test_results_aggregates_bindable = ObjectType("TestResultsAggregates")
+from utils.test_results import get_results
 
 
-class TestResultsAggregates(TypedDict):
+@dataclass
+class TestResultsAggregates:
     total_duration: float
-    total_duration_percent_change: float | None
     slowest_tests_duration: float
-    slowest_tests_duration_percent_change: float | None
     total_slow_tests: int
-    total_slow_tests_percent_change: float | None
     fails: int
-    fails_percent_change: float | None
     skips: int
-    skips_percent_change: float | None
+    total_duration_percent_change: float | None = None
+    slowest_tests_duration_percent_change: float | None = None
+    total_slow_tests_percent_change: float | None = None
+    fails_percent_change: float | None = None
+    skips_percent_change: float | None = None
+
+
+def calculate_aggregates(table: pl.DataFrame) -> pl.DataFrame:
+    return table.select(
+        (
+            pl.col("avg_duration")
+            * (pl.col("total_pass_count") + pl.col("total_fail_count"))
+        )
+        .sum()
+        .alias("total_duration"),
+        (
+            pl.when(pl.col("avg_duration") >= pl.col("avg_duration").quantile(0.95))
+            .then(
+                pl.col("avg_duration")
+                * (pl.col("total_pass_count") + pl.col("total_fail_count"))
+            )
+            .otherwise(0)
+            .top_k(100)
+            .sum()
+            .alias("slowest_tests_duration")
+        ),
+        (pl.col("total_skip_count").sum()).alias("skips"),
+        (pl.col("total_fail_count").sum()).alias("fails"),
+        ((pl.col("avg_duration") >= pl.col("avg_duration").quantile(0.95)).sum()).alias(
```

Comment:
do we still need to do the limit 100 here?

Reply:
indeed, that's being handled above but not here